File: Top Interview Questions/Easy Collection/Best Time to Buy and Sell Stock II.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        profit = 0
        flag = 0
        buy = 10001
        sell = -1
        for price in prices:
            if flag == 0:
                if price <= buy:
                    buy = price
                else:
                    flag = 1
                    sell = buy
            if flag == 1:
                if price > sell:
                    sell = price
                else:
                    profit += sell - buy
                    buy = price
                    flag = 0
        if flag == 1:
            profit += sell - buy
        return profit

File: Top Interview Questions/Easy Collection/C/Array/Best Time to Buy and Sell Stock II.c
int maxProfit(int* prices, int pricesSize){
    if(pricesSize == 1)
        return 0;
    
    int profit = 0;
    int buy = 30001;
    for(int i=0; i<pricesSize; i++){
        if(prices[i] > buy)
            profit += prices[i] - buy;
        buy = prices[i];
    }
    return profit;
}